tutorials/groups.qmd

```diff
@@ -0,0 +1,242 @@
+---
+title: work with groups
+toc: true
+---
+
+```{julia}
+#| echo: false
+#| code-fold: true
+#| output: false
+using Pkg;
+cd(@__DIR__)
+Pkg.activate("."); # for reproducibility use the local tutorial environment.
+Pkg.develop(path="../") # a trick to work on the local dev version
+using Markdown
+```
+
+This is a short overview of group support in [`Manifolds.jl`](https://juliamanifolds.github.io/Manifolds.jl/) and how to get started working with them.
+
+Groups currently available in `Manifolds.jl` are listed in ``[group section](@ref GroupManifoldSection)``{=commonmark}.
+
+You can read more about the theory of Lie groups for example in [Chirikjian:2012](@cite).
+An example application of Lie groups in robotics is described in [SolaDerayAtchuthan:2021](@cite).
+
+First, let's load libraries we will use.
+`RecursiveArrayTools` is necessary because `ArrayPartition` defined in it is used as one of the possible representations of elements of product and semidirect product groups.
+`StaticArrays` can be used to speed up some operations.
+
+```{julia}
+using Manifolds, RecursiveArrayTools, StaticArrays
+```
+
+## Introduction: group of rotations on a plane
+
+Let's first consider an example of the group of rotations of a plane, $\operatorname{SO}(2)$.
+They can be represented in several ways, for example as angles of rotation (which corresponds to [`RealCircleGroup`](@ref)), unit complex numbers ([`CircleGroup`](@ref)), or rotation matrices ([`SpecialOrthogonal`](@ref)).
+Let's consider the last representation since it is the most nontrivial one and can be more easily generalized to other groups.
+
+The associated manifolds and groups are defined by:
+
+```{julia}
+G = SpecialOrthogonal(2)
+M = base_manifold(G)
+@assert M === Rotations(2)
+```
+
+This duality (Lie group and the underlying manifold being separate) is a common pattern in `Manifolds.jl`. The group `G` can be used for both Lie group-specific operations and metric-specific operation, while the manifold `M` only allows using manifold and metric operations. This way groups can be specialized in ways not relevant to plain manifolds, and if someone doesn't use the groups structure, they don't have to consider it by just using the manifold.
+
+Some basic definitions
+
+```{julia}
+# default basis
+B = DefaultOrthogonalBasis()
+# Identity rotation
+p0 = @SMatrix [1.0 0; 0 1]
+
+# Group identity element of a special type
+IG = Identity(G)
+```
+
+
+Let's say we want to define a manifold point `p_i` some rotation θ from the [`identity_element`](@ref) reference rotation `p0` (another point on the manifold that we will use as reference)
+
+```{julia}
+# + radians rotation from x-axis on plane to point i
+xθi = π/6
+```
+
+### From Coordinates
+
+To get our first Lie algebra element we can use the textbook [`hat`](@ref), or equivaliently a more generalized [`get_vector`](@ref), function:
```

Comment:
I do not understand the phrase “the textbook” here. What does this refer to?

Reply:
I will rephrase it.